co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/carlaconnect/CarlaMessageReceiver.java

```diff
@@ -0,0 +1,104 @@
+/*
+*Copyright (C) 2023 LEIDOS.
+*
+*Licensed under the Apache License, Version 2.0 (the "License"); you may not
+*use this file except in compliance with the License. You may obtain a copy of
+*the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing, software
+*distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+*WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+*License for the specific language governing permissions and limitations under
+*the License.
+*/
+package org.eclipse.mosaic.fed.carla.carlaconnect;
+
+import java.net.URL;
+import java.net.MalformedURLException;
+import org.apache.xmlrpc.XmlRpcException;
+import org.apache.xmlrpc.client.XmlRpcClient;
+import org.apache.xmlrpc.client.XmlRpcClientConfigImpl;
+import org.eclipse.mosaic.fed.carla.ambassador.CarlaAmbassador;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a class uses xmlrpc to connect with CARLA CDASim adapter service
+ */
+public class CarlaMessageReceiver implements Runnable {
```

Comment:
Rename to CarlaXmlRpcClient

Reply:
Also why are we making this class implement runnable. It is not an open socket from which we plan to constantly consume information. Instead it is a client which can be used to make event driven requests. Lets remove Runnable here and just implement methods that execute a request and return information, and maybe a single initialize or connect method that establishes a connection.